api-2.2/src/main/java/org/openmrs/module/fhir/helper/ConditionHelperImpl2_2.java

```diff
@@ -68,6 +67,48 @@ public Condition createCondition(Condition condition) {
 		return generateFHIRCondition(openMrsCondition);
 	}
 
+	/**
+	 * @see org.openmrs.module.fhir.api.helper.ConditionHelper#updateFHIRCondition(org.hl7.fhir.dstu3.model.Condition)
+	 */
+	@Override
+	public Condition updateFHIRCondition(Condition condition) {
+		if (condition.getId() != null) {
+			ConditionService conditionService = Context.getService(ConditionService.class);
+			org.openmrs.Condition retrievedCondition = conditionService.getConditionByUuid(condition.getId());
+			return retrievedCondition != null ? updateRetrievedCondition(condition, retrievedCondition) :
+					createCondition(condition);
+		} else {
+			throw new UnprocessableEntityException("condition Id cannot be null");
+		}
+
+	}
+
+	/**
+	 * Update retrieved condition
+	 * @param condition FHIR condition to update
```

Comment:
What is the difference between `the condition to update` and `the condition to be updated`?

Reply:
condition to update is the condition from the user request and condition to be updated is the existing condition retrieved from DB

Reply:
I find that confusing. Can you user different names or terms? condition to update means exactly the same as condition to be updated.